includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -1412,6 +1403,11 @@ protected function calculate_shipping( $address = [] ) {
 			}
 		}
 
+		// Remove subscription shipping package filter if there is free trial in the cart to allow the calculation of shipping costs.
+		if ( $this->cart_contains_free_trial() ) {
+			remove_filter( 'woocommerce_cart_shipping_packages', 'WC_Subscriptions_Cart::set_cart_shipping_packages', -10, 1 );
+		}
```

Comment:
Question: what should happen if the cart contains a free shippable trial and a free non-shippable trial?

While that filter currently handles only filtering out free trials, it might change in future. I think it may be worth considering one of the following two approaches to this:
 * Fix this upstream in WooCommerce Subscriptions -- that code should not filter out shippable free trials
 * Change the implementation here so that we run a filter at an earlier priority that identifies any shippable free trials, then runs the filter above, and then runs a later filter that reinstates the shippable free trials

We may need to tackle those things in a follow-up, but I think we should think through the edge cases.